basil/HL/tektronix_oscilloscope.py

```diff
@@ -0,0 +1,197 @@
+            WaveType.ANALOG: lambda x: x,
+        }
+
+        results = {}
+        sources = ['CH' + str(channel)]
+        for source in sources:
+            if source.split("_")[0] not in results:
+                # Determine the type of waveform and set key interface parameters
+                wave_type = self._classify_waveform(source)
+                channel = channel_table[wave_type](source)
+                encoding, bit_nr, datatype = encoding_table[wave_type]
+
+                # Set the start and stop point of the record
+                rec_len = int(self._intf.query("horizontal:recordlength?").strip())
+
+                # Keep track of each super channel and math source that has been handled
+                results[source.split("_")[0]] = JobParameters(
+                    wave_type, channel, encoding, bit_nr, datatype, rec_len
+                )
+        return results
+
+    def _classify_waveform(self, source):
+        if source[0:4] == "MATH":
+            wave_type = WaveType.MATH
+        elif "_" in source:
+            wave_type = WaveType.DIGITAL
+        else:
+            wave_type = WaveType.ANALOG
+        return wave_type
+
+    def _get_xscale(self):
+        """Get the horizontal scale of the instrument"""
+
+        # This function will generate a VISA timeout if there is no waveform data
+        # available and the channel is enabled.
+        result = None
+        try:
+            xincr = self._intf.query("WFMOutpre:XINCR?").strip()
+        except Exception:
+            pass
+        else:
+            # collect more horizontal data
+            pt_off = self._intf.query("WFMOutpre:PT_OFF?").strip()
+            xzero = self._intf.query("WFMOutpre:XZERO?").strip()
+            xunit = self._intf.query("WFMOutpre:XUNIT?").strip()
+
+            # calculate horizontal scale
+            slope = float(xincr)
+            offset = float(pt_off) * -slope + float(xzero)
+            unit = xunit.strip('"')
+            result = XScale(slope, offset, unit)
+        return result
+
+    def _get_yscale(self, channel=1):
+        scale = float(self._intf.query("{}:SCALE?".format('CH' + str(channel))))
+        position = float(self._intf.query("{}:POSITION?".format('CH' + str(channel))))
+        top = scale * (5 - position)
+        bottom = scale * (-5 - position)
+        return YScale(top=top, bottom=bottom)
+
+    def _has_data_available(self, source):
+        """Checks that the source seems to have actual data available for download.
+        This function should be called before performing a curve query."""
+        # Use "display:global:CH{x}:state?" to determine if the channel is displayed
+        # and available for download
+        if source == "NONE":
+            display_on = False
+        else:
+            display_on = bool(
+                int(self._intf.query("display:global:{}:state?".format(source)))
+            )
+        return display_on
```

Comment:
A more pythonic solution to implement this would be to return immediately, like:

    def has_data_available(self, source):
        if source == 'None':
            return False
        return bool(int(self._intf.query("display:global:{}:state?".format(source))))